src/main/java/seedu/address/model/attendance/SessionList.java

```diff
@@ -47,19 +103,110 @@ public boolean contains(Session toCheck) {
+            s.updateSessionAfterDelete(studentId, internalPersonList);
+        }
+    }
+
+    /**
+     * Updates all sessions after adding a student
+     */
+    public void updateAllSessionsAfterAdd() {
+        for (Session s : sessions) {
+            s.updateSessionAfterAdd(internalPersonList);
+        }
+    }
+
+    /**
+     * Finds the session using the given {@code sessionName} and students' participation status
+     * according to the {@code indexRange}
+     */
+    public void updateStudentParticipation(SessionName sessionName, IndexRange indexRange) {
+        requireNonNull(sessionName);
+        requireNonNull(indexRange);
+
+        for (Session session: sessions) {
+            if (session.getSessionName().equals(sessionName)) {
+                session.updateParticipation(indexRange);
+            }
+        }
+    }
+
+    /**
+     * Finds the session using the given {@code sessionName} and students' presence status
+     * according to the {@code indexRange}
+     */
+    public void updateStudentPresence(SessionName sessionName, IndexRange indexRange) {
+        requireNonNull(sessionName);
+        requireNonNull(indexRange);
+
+        for (Session session: sessions) {
+            if (session.getSessionName().equals(sessionName)) {
+                session.updatePresence(indexRange);
+            }
         }
     }
 
+    /**
+     * Clears all the existing sessions in the session list.
+     */
+    public void clearSessions() {
+        sessions.clear();
+    }
+
+    @Override
+    public SortedSet<Session> getSessions() {
+        return sessions;
+    }
+
+    @Override
+    public ObservableList<Person> getInternalPersonList() {
+        return internalPersonList;
+    }
+
+    @Override
+    public Iterator<Session> iterator() {
+        return sessions.iterator();
+    }
+
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof SessionList // instanceof handles nulls
-                && sessions.equals(((SessionList) other).sessions)
-                && masterList.equals(((SessionList) other).masterList));
+        if (other instanceof SessionList) {
+
+            if (sessions.size() != ((SessionList) other).getSessions().size()) {
+                return false;
+            }
+
+            Session[] testArray1 = new Session[sessions.size()];
+            Session[] testArray2 = new Session[sessions.size()];
+
+            int i = 0;
+            int j = 0;
+            for (Session s: sessions) {
+                testArray1[i] = s;
+                i++;
+            }
+            for (Session s: ((SessionList) other).getSessions()) {
+                testArray2[j] = s;
+                j++;
+            }
+
+            for (int k = 0; k < sessions.size(); k++) {
+                if (!testArray1[k].isSameSession(testArray2[k])) {
+                    return false;
+                }
+            }
```

Comment:
Maybe some comments could be added here to more clearly explain the thought process?